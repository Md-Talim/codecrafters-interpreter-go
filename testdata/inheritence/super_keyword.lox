// Test Case 1
class Doughnut {
  cook() {
    print "Fry until golden brown.";
  }
}

// Super can be used to call the overridden method
// of the parent class
class BostonCream < Doughnut {
  cook() {
    super.cook();
  }
}

BostonCream().cook();

// Test Case 2
class Base {
  method() {
    print "Base.method()";
  }
}

class Parent < Base {
  method() {
    super.method();
  }
}

class Child < Parent {
  method() {
    super.method();
  }
}

var parent = Parent();
parent.method();
var child = Child();
child.method();