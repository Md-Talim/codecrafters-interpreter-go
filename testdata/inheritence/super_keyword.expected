Fry until golden brown.
Base.method()
Base.method()